src/pymachine/definition_parser.py

```diff
@@ -487,84 +472,78 @@ def __parse_expr(self, expr, root, loop_to_defendum=True,
         logging.debug(expr)
         raise pe
 
-    def __parse_definition(self, definition, root, loop_to_defendum=True,
-                           three_parts=False):
+    def __parse_definition(self, definition, root, loop_to_defendum=True):
         logging.debug(str(definition))
         for d in definition:
-            yield self.__parse_expr(d, root, loop_to_defendum, three_parts)[0]
+            yield self.__parse_expr(d, root, loop_to_defendum)[0]
 
-    def parse_into_machines(self, string, printname_index=0, add_indices=False,
-                            loop_to_defendum=True, three_parts=False):
-        printname = string.split('\t')[printname_index]
-        try:
-            id_, urob, pos, def_, comment = string.split('\t')[4:]
-        except:
-            raise Exception(string.split('\t'))
+    def parse_into_machines(self, printname, id_, def_, add_indices=False,
+                            loop_to_defendum=True):
 
         machine = self.create_machine(printname.lower(), 1)
-        #TODO =AGT -> partition 1, =PAT -> partition 2, =TO -> ?
 
         if add_indices:
             machine.printname_ = machine.printname() + id_sep + id_
 
-        if def_ != '':
-            logging.debug(def_)
-            parsed = self.parse(def_)
-            logging.debug(parsed)
-            for parsed_expr in self.__parse_definition(
-                    parsed[0], machine, loop_to_defendum, three_parts):
-                machine.append(parsed_expr, 0)
+        #if def_ != '':
+        logging.debug(def_)
+        parsed = self.parse(def_)
+        logging.debug(parsed)
+        for parsed_expr in self.__parse_definition(
+                parsed[0], machine, loop_to_defendum):
+            machine.append(parsed_expr, 0)
 
         self.unify(machine)
         return machine
 
-def read(f, plur_filn, printname_index=0, add_indices=False,
```

Comment:
I don't really see the need for 1) and 3), but if you do, go ahead. But
then I think you should check all code in pymachine and 4lang to make sure
that everything keeps working, and then create PRs in both repos which can
be merged simultaneously.

On Thu, Mar 31, 2016 at 4:21 PM, Márton Makrai <notifications@github.com>
wrote:

> In src/pymachine/definition_parser.py
> <https://github.com/kornai/pymachine/pull/8#discussion_r58061528>:
>
> >
> >          self.unify(machine)
> >          return machine
> >
> > -def read(f, plur_filn, printname_index=0, add_indices=False,
>
> I didn't realize that other modules use it, thank you for checking. I
> don't want to remove it but to make some changes to it. I go into details
> to discuss what I should do to this request.
>
>    1. I would like to rename the function, because read is a built in
>    function. I found two modules importing this function, ./wrapper.py and
>    4lang/src/fourlang/lexicon.py. Both import it as read_defs. This name would
>    be perfect in my opinion. What do you think? (Btw do we want to use these
>    modules? Are there other modules importing this function that we (plan to)
>    use?)
>    2. Arguments: by now, the second argument is plur_filn, which is
>    removed in this request (this is the point of the request). (The other
>    change is less important: In the present form of the request,
>    DefinitionParser is instantiated out of this function, and is passed the
>    parser as a second argument.)
>    3. Return: By now, this function has returned a dictionary, {printname
>    : set of machines} (one-element sets, if I'm not mistaken). I would prefer
>    yielding machines, probably (printname, machine) pairs. Would it fit to
>    what you are doing now?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kornai/pymachine/pull/8/files/cc47e5dce3b0c81bb3d2f7c197a0af055c71939c#r58061528>
>